_templates/l10n.php

```diff
@@ -35,334 +37,368 @@ function list_langs() {
         'zh_CN' => '简体中文',
         'zh_TW' => '繁體中文',
     );
-}
 
-function lang_dir($lang) {
-    return dirname(__FILE__).'/../lang/'.$lang;
-}
+    protected $lang = 'en';
 
-function is_lang($lang) {
-    if (!is_string($lang)) {
-        return false;
-    }
-    if (!preg_match('#^[a-z]{2}(_([A-Z]{2}|[A-Z][a-z]+))?$#', $lang)) {
-        return false;
+    protected $domain = null;
+
+    public function __construct($lang = null) {
+        if (empty($lang)) {
+            $lang = $this->get_page_lang();
+        }
+        $this->lang = $lang;
     }
-    if ($lang == 'en') {
-        return true;
+
+    public function init() {
+        global $sitewide, $page; // Global site variables
```

Comment:
You started to create this class after our discussion about the current procedural code which abuses of globals all over the place which makes it hard to test.

Unit testing should be done in isolation and when your unit depends on globals and super globals is hard to test it. Its not impossible, but you will end up with a pretty long bootstrap for that test.

Your unit should know that it needs param one, two and three in order to work and that's it. It should not care from where those parameters came from.